publication/models/account_analytic_invoice_line.py

```diff
@@ -26,3 +26,40 @@ def action_distribution_list(self):
         action['view_mode'] = 'form'
         action['target'] = 'current'
         return action
+
+    @api.model
+    def create(self, vals):
+        result = super(AccountAnalyticInvoiceLine, self).create(vals)
+        if result.publication:
+            self.env['distribution.list'].create({
+                'product_id': result.product_id.id,
+                'contract_partner_id':
+                result.analytic_account_id.partner_id.id,
+                'partner_id': result.partner_id.id,
+                'date_start': result.date_start,
+                'analytic_invoice_line_id': result.id,
```

Comment:
Leave out this extra field. But you will have to use the quantity of the contract line for the field 'copies'.

Reply:
@NL66278 So if a distribution list does not map to a `analytic_invoice_line`, when the user marks an `analytic_invoice_line` as a publication, what do I do on the distribution lists assigned to this partner? Do I add a new one or modify an already existing one if existing? With what parameters? Dates and product?

Reply:
See also comments below, our comments crossed.

There should be only one, or rather at most one, entry in the distribution list for the combination of contract partner and product (publication).